docs/install/debug.md

```diff
@@ -126,3 +126,93 @@ target remote | /usr/local/lib/valgrind/../../bin/vgdb --pid=31925
 Every press of the 'c' key in the gdb shell will move to the location of
 the next valgrind error.
 
+
+Profiling and performance benchmarking
+--------------------------------------
+
+NEURON has recently gained built-in support for performance profilers. Many modern profilers provide APIs for instrumenting code. This allows the profiler to enable timers or performance counters and store results into appropriate data structures. For implementation details of the generic profiler interface check `src/utils/profile/profiler_interface.h` NEURON now supports following profilers:
+
+* [Caliper](https://software.llnl.gov/Caliper/)
+* [likwid](https://github.com/RRZE-HPC/likwid)
+* [tau](https://www.paratools.com/tau)*
+* [craypat](https://docs.nersc.gov/tools/performance/craypat/)*
+
+*to use this profiler some additional changes to the CMake files might be needed.
+
+#### Instrumentation
+
+NEURONs code has been already instrumented with instrumentor regions in many performance-critical functions of the code. The existing regions have been given the same names as in CoreNEURON to allow side-by-side comparision when running simulations with and without CoreNEURON enabled. More regions can easily be added into the code in one of two ways:
+
+1. using calls to `phase_begin()`, `phase_end()`
+
+```c++
+void some_function() {
+    nrn::Instrumentor::phase_begin("some_function");
+    // code to be benchmarked
+    nrn::Instrumentor::phase_end("some_function");
+}
+```
+
+2. using scoped automatic variables
+
+```c++
+void some_function() {
+    // unrelated code
+    {
+        nrn::Instrumentor::phase p("critical_region");
+        // code to be benchmarked
+    }
+    // more unrelated code
+}
+```
+Note: Don't forget to include the profiler header in the respective source files.
+
+#### Running benchmarks
+
+To enable a profiler, one needs to rebuild NEURON with the appropriate flags set. Here is how one would build NEURON with Caliper enabled:
```

Comment:
pure `nitpick`: A link to GitHub-CI yaml file where we install Caliper and then mentioning option in below cmake `-DCMAKE_PREFIX_PATH=/path/to/caliper/share/cmake/caliper` would be helpful.

Reply:
I don't think CMAKE_PREFIX_PATH would be the right variable to set, though

Reply:
Totally OK with not having an extra reference on how to install Caliper. 


> I don't think CMAKE_PREFIX_PATH would be the right variable to set, though

Yeah, because in the the GH coverage.yml you install it system-wide.

Reply:
my bad, that is totally the right variable to set 😅 , I must have mixed it up with another one, there are only like 3 or 4 different CMake variables, right?
Anyway, I can add that variable into the tutorial, but how about we don't mention the coverage.yml here? I don't see how this is relevant.
@alexsavulescu , yes I installed system wide, because for the CI, who cares 😄

Reply:
> Anyway, I can add that variable into the tutorial, but how about we don't mention the coverage.yml here? I don't see how this is relevant.

👌 👍